js_modules/dagster-ui/packages/ui-core/src/overview/OverviewTimelineRoot.tsx

```diff
@@ -79,7 +80,10 @@ export const OverviewTimelineRoot = ({Header, TabButton}: Props) => {
     [hourWindow, now, offsetMsec],
   );
 
-  const {jobs, initialLoading, refreshState} = useRunsForTimeline(range);
+  const runsForTimelineRet = useRunsForTimeline(range);
+
+  // Use deferred value to allow paginating quickly with the UI feeling more responsive.
+  const {jobs, initialLoading, refreshState} = useDeferredValue(runsForTimelineRet);
```

Comment:
This is actually pretty neat, you guys should try paginating without the `useDeferredValue` and you'll be able to see the difference in responsiveness